lib/Dialect/FIRRTL/CMakeLists.txt

```diff
@@ -13,9 +13,10 @@ add_circt_dialect_library(MLIRFIRRTL
   Support
 
   LINK_LIBS PUBLIC
+  MLIRRTL
```

Comment:
@lattner after rebasing, I found that I needed this extra dependency to link. Is this dependency valid? I think it was needed for this:

https://github.com/llvm/circt/commit/73dbad3377c3fc17bdbc94c926c4d7e3023f6ff8#diff-6be8f16b249664b12c96f16587df5fbae2167725d4df3b2d04e075610fcbc9c9R1329

Reply:
Ah, that makes sense.  You're correct, but this is super annoying.  It would be better for FIRRTL to not depend on RTL like this (lowertortl would of course) this LGTM, [I filed a bug](https://github.com/llvm/circt/issues/346).